src/app/static/src/app/store/baseline/actions.ts

```diff
@@ -134,21 +133,36 @@ export const actions: ActionTree<BaselineState, DataExplorationState> & Baseline
 
     async importPJNZ(context, url) {
         if (url) {
-            await uploadOrImportPJNZ(context, {url: "/adr/pjnz/", payload: qs.stringify({url})},
+            const data = {
+                url,
+                id: context.state.selectedDataset?.id || "",
+                resourceId: context.state.selectedDataset?.resources?.pjnz?.id || ""
+            }
+            await uploadOrImportPJNZ(context, {url: "/adr/pjnz/", payload: data},
                 getFilenameFromImportUrl(url));
         }
     },
 
     async importPopulation(context, url) {
         if (url) {
-            await uploadOrImportPopulation(context, {url: "/adr/population/", payload: qs.stringify({url})},
+            const data = {
+                url,
+                id: context.state.selectedDataset?.id || "",
+                resourceId: context.state.selectedDataset?.resources?.pop?.id || ""
+            }
```

Comment:
It might be nice to pull the adr resource payload contruction out into its own method, as this logic is repeated per resource type.

Reply:
I am not sure repetition can be avoided, resource, url are all unique.

Reply:
I just meant you could have something like
```
const buildData = (url: string, resourceType: string) => ({
    url,
    id: context.state.selectedDataset?.id || "",
    resourceId: context.state.selectedDataset?.resources?[resourceType]?.id || ""
});
...
...
const data = buildData(url, "pop");
```
which would save repetition of drilling down to get dataset id and resource ids

Reply:
OK, this works for me:
```
const buildData = (context: ActionContext<BaselineState, DataExplorationState>, url: string, resourceType: DatasetResourceType) => ({
    url,
    id: context.state.selectedDataset?.id || "",
    resourceId: context.state.selectedDataset?.resources?.[resourceType]?.id || ""
});
```
if you declare the DatasetResourceType in types.ts:
`export type DatasetResourceType = "pjnz" | "pop"  | "program" | "anc" | "shape" | "survey";`